libs/agno/agno/memory/manager.py

```diff
@@ -712,6 +717,166 @@ def _get_first_n_memories(self, user_id: str, limit: Optional[int] = None) -> Li
 
         return sorted_memories_list
 
+    def optimize_memories(
+        self,
+        user_id: Optional[str] = None,
+        strategy: Union[
+            MemoryOptimizationStrategyType, MemoryOptimizationStrategy
+        ] = MemoryOptimizationStrategyType.SUMMARIZE,
```

Comment:
We should probably be ready to receive a list here.

Reply:
this would mean to accept multiple stratergies in the param. idts that would be ideal and we should be doing this.
Not sure how we will optimise memories by using multiple stratergies, we should accept only one stratergy to do optimisation.

Reply:
I don't think that we should be receiving a list over here because the optimize function works on a single strategy. Receiving a list means that we are accepting multiple strategies for optimizing memories. Not sure how we will handle optimization with multiple strategies. In my opinion, keeping it to a single strategy will be the right choice.